diff -up grub-2.06/docs/grub.texi.40~ grub-2.06/docs/grub.texi
--- grub-2.06/docs/grub.texi.40~	2022-06-26 18:26:01.581138878 +0200
+++ grub-2.06/docs/grub.texi	2022-06-26 18:26:01.614471683 +0200
@@ -1411,6 +1411,13 @@ entry for recovery mode.  This option li
 only to the default menu entry, after those listed in
 @samp{GRUB_CMDLINE_LINUX}.
 
+@item GRUB_CMDLINE_LINUX_RECOVERY
+Unless @samp{GRUB_DISABLE_RECOVERY} is set to @samp{true}, two menu
+entries will be generated for each Linux kernel: one default entry and one
+entry for recovery mode. This option lists command-line arguments to add
+only to the recovery menu entry, before those listed in @samp{GRUB_CMDLINE_LINUX}.
+The default is @samp{single}.
+
 @item GRUB_CMDLINE_NETBSD
 @itemx GRUB_CMDLINE_NETBSD_DEFAULT
 As @samp{GRUB_CMDLINE_LINUX} and @samp{GRUB_CMDLINE_LINUX_DEFAULT}, but for
diff -up grub-2.06/util/grub.d/10_linux.in.40~ grub-2.06/util/grub.d/10_linux.in
--- grub-2.06/util/grub.d/10_linux.in.40~	2022-06-26 18:26:01.394475173 +0200
+++ grub-2.06/util/grub.d/10_linux.in	2022-06-26 18:27:44.322876542 +0200
@@ -49,6 +49,10 @@ case ${GRUB_DEVICE} in
   ;;
 esac
 
+if [ "x${GRUB_CMDLINE_LINUX_RECOVERY}" = "x" ] ; then
+  GRUB_CMDLINE_LINUX_RECOVERY=single
+fi
+
 # Default to disabling partition uuid support to maintian compatibility with
 # older kernels.
 GRUB_DISABLE_LINUX_PARTUUID=${GRUB_DISABLE_LINUX_PARTUUID-true}
@@ -323,7 +327,7 @@ while [ "x$list" != "x" ] ; do
 		"${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT} plymouth.enable=0 systemd.unit=multi-user.target"
   if [ "x${GRUB_DISABLE_RECOVERY}" != "xtrue" ]; then
     linux_entry "${OS}" "${version}" recovery \
-                "single ${GRUB_CMDLINE_LINUX_RECOVERY}"
+                "${GRUB_CMDLINE_LINUX_RECOVERY} ${GRUB_CMDLINE_LINUX_RECOVERY}"
   fi
 
   list=`echo $list | tr ' ' '\n' | fgrep -vx "$linux" | tr '\n' ' '`
diff -up grub-2.06/util/grub.d/20_linux_xen.in.40~ grub-2.06/util/grub.d/20_linux_xen.in
--- grub-2.06/util/grub.d/20_linux_xen.in.40~	2020-12-23 00:24:17.000000000 +0100
+++ grub-2.06/util/grub.d/20_linux_xen.in	2022-06-26 18:26:01.614471683 +0200
@@ -43,6 +43,10 @@ case ${GRUB_DEVICE} in
   ;;
 esac
 
+if [ "x${GRUB_CMDLINE_LINUX_RECOVERY}" = "x" ] ; then
+  GRUB_CMDLINE_LINUX_RECOVERY=single
+fi
+
 # Default to disabling partition uuid support to maintian compatibility with
 # older kernels.
 GRUB_DISABLE_LINUX_PARTUUID=${GRUB_DISABLE_LINUX_PARTUUID-true}
@@ -343,7 +347,7 @@ while [ "x${xen_list}" != "x" ] ; do
 	    "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT}" "${GRUB_CMDLINE_XEN} ${GRUB_CMDLINE_XEN_DEFAULT}"
 	if [ "x${GRUB_DISABLE_RECOVERY}" != "xtrue" ]; then
 	    linux_entry "${OS}" "${version}" "${xen_version}" recovery \
-		"single ${GRUB_CMDLINE_LINUX}" "${GRUB_CMDLINE_XEN}"
+		"${GRUB_CMDLINE_LINUX_RECOVERY} ${GRUB_CMDLINE_LINUX}" "${GRUB_CMDLINE_XEN}"
 	fi
 
 	list=`echo $list | tr ' ' '\n' | fgrep -vx "$linux" | tr '\n' ' '`
diff -up grub-2.06/util/grub-mkconfig.in.40~ grub-2.06/util/grub-mkconfig.in
--- grub-2.06/util/grub-mkconfig.in.40~	2022-06-26 18:26:01.347809247 +0200
+++ grub-2.06/util/grub-mkconfig.in	2022-06-26 18:26:01.614471683 +0200
@@ -231,6 +231,7 @@ export GRUB_DEFAULT \
   GRUB_DISTRIBUTOR \
   GRUB_CMDLINE_LINUX \
   GRUB_CMDLINE_LINUX_DEFAULT \
+  GRUB_CMDLINE_LINUX_RECOVERY \
   GRUB_CMDLINE_XEN \
   GRUB_CMDLINE_XEN_DEFAULT \
   GRUB_CMDLINE_LINUX_XEN_REPLACE \
