From f1f912f9c1b096ec2fa97c4de0c5376fdd860a17 Mon Sep 17 00:00:00 2001
From: Elyes Haouas <ehaouas@noos.fr>
Date: Fri, 4 Mar 2022 07:41:58 +0100
Subject: [PATCH 126/261] font: Remove trailing whitespaces

Signed-off-by: Elyes Haouas <ehaouas@noos.fr>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>
---
 grub-core/font/font.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/grub-core/font/font.c b/grub-core/font/font.c
index d09bb38d8..42189c325 100644
--- a/grub-core/font/font.c
+++ b/grub-core/font/font.c
@@ -936,7 +936,7 @@ grub_font_get_height (grub_font_t font)
 }
 
 /* Get the glyph for FONT corresponding to the Unicode code point CODE.
-   Returns the ASCII glyph for the code if no other fonts are available. 
+   Returns the ASCII glyph for the code if no other fonts are available.
    The glyphs are cached once loaded.  */
 struct grub_font_glyph *
 grub_font_get_glyph (grub_font_t font, grub_uint32_t code)
-- 
2.36.1

