From a4c3ca4474ce295f696e27778020a6e3b05c1be4 Mon Sep 17 00:00:00 2001
From: Glenn Washburn <development@efficientek.com>
Date: Tue, 12 Apr 2022 16:56:18 +0000
Subject: [PATCH 199/261] docs: Add note that drivemap is only available on
 i386-pc

Signed-off-by: Glenn Washburn <development@efficientek.com>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>
---
 docs/grub.texi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/docs/grub.texi b/docs/grub.texi
index 62e51564b..93f2d311e 100644
--- a/docs/grub.texi
+++ b/docs/grub.texi
@@ -4448,6 +4448,8 @@ For example:
 @example
 drivemap -s (hd0) (hd1)
 @end example
+
+NOTE: Only available on i386-pc.
 @end deffn
 
 
-- 
2.36.1

