From 4eb77428abc0140757c425816dee5f7e85a38b8e Mon Sep 17 00:00:00 2001
From: Fedora Ninjas <grub2-owner@fedoraproject.org>
Date: Mon, 8 Jun 2020 06:50:21 +0200
Subject: [PATCH] fix build with rpm-4.16

---
 configure.ac | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/configure.ac b/configure.ac
index 6c46756..2fbee9a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1887,6 +1887,15 @@ if test x"$rpm_sort_excuse" = x ; then
              [Define to 1 if you have the rpm library.])
 fi
 
+if test x"$LIBRPM" = x ; then
+  # Check for rpm library.
+  AC_CHECK_LIB([rpmio], [rpmvercmp], [],
+               [rpm_sort_excuse="rpmio missing rpmvercmp"])
+   LIBRPM="-lrpmio";
+   AC_DEFINE([HAVE_RPMIO], [1],
+             [Define to 1 if you have the rpm library.])
+fi
+
 AC_SUBST([LIBRPM])
 
 LIBGEOM=
-- 
2.27.0

