diff --git a/grub-core/boot/i386/pc/boot.S b/grub-core/boot/i386/pc/boot.S
index 2bd0b2d..ea167fe 100644
--- a/grub-core/boot/i386/pc/boot.S
+++ b/grub-core/boot/i386/pc/boot.S
@@ -249,9 +249,6 @@ real_start:
 	/* save drive reference first thing! */
 	pushw	%dx
 
-	/* print a notification message on the screen */
-	MSG(notification_string)
-
 	/* set %si to the disk address packet */
 	movw	$disk_address_packet, %si
 
diff --git a/grub-core/boot/i386/pc/diskboot.S b/grub-core/boot/i386/pc/diskboot.S
index c1addc0..68d31de 100644
--- a/grub-core/boot/i386/pc/diskboot.S
+++ b/grub-core/boot/i386/pc/diskboot.S
@@ -50,11 +50,6 @@ _start:
 	/* save drive reference first thing! */
 	pushw	%dx
 
-	/* print a notification message on the screen */
-	pushw	%si
-	MSG(notification_string)
-	popw	%si
-
 	/* this sets up for the first run through "bootloop" */
 	movw	$LOCAL(firstlist), %di
 
diff --git a/grub-core/gettext/gettext.c b/grub-core/gettext/gettext.c
index edebed9..4263d76 100644
--- a/grub-core/gettext/gettext.c
+++ b/grub-core/gettext/gettext.c
@@ -439,16 +439,12 @@ static char *
 grub_gettext_env_write_lang (struct grub_env_var *var
 			     __attribute__ ((unused)), const char *val)
 {
-  grub_err_t err;
+  grub_err_t __attribute__((__unused__)) err;
   err = grub_gettext_init_ext (&main_context, val, grub_env_get ("locale_dir"),
 			       grub_env_get ("prefix"));
-  if (err)
-    grub_print_error ();
 
   err = grub_gettext_init_ext (&secondary_context, val,
 			       grub_env_get ("secondary_locale_dir"), 0);
-  if (err)
-    grub_print_error ();
 
   return grub_strdup (val);
 }
@@ -456,23 +452,19 @@ grub_gettext_env_write_lang (struct grub_env_var *var
 void
 grub_gettext_reread_prefix (const char *val)
 {
-  grub_err_t err;
+  grub_err_t __attribute__((__unused__)) err;
   err = grub_gettext_init_ext (&main_context, grub_env_get ("lang"),
 			       grub_env_get ("locale_dir"),
 			       val);
-  if (err)
-    grub_print_error ();
 }
 
 static char *
 read_main (struct grub_env_var *var
 	   __attribute__ ((unused)), const char *val)
 {
-  grub_err_t err;
+  grub_err_t __attribute__((__unused__)) err;
   err = grub_gettext_init_ext (&main_context, grub_env_get ("lang"), val,
 			       grub_env_get ("prefix"));
-  if (err)
-    grub_print_error ();
   return grub_strdup (val);
 }
 
@@ -480,12 +472,9 @@ static char *
 read_secondary (struct grub_env_var *var
 		__attribute__ ((unused)), const char *val)
 {
-  grub_err_t err;
+  grub_err_t __attribute__((__unused__)) err;
   err = grub_gettext_init_ext (&secondary_context, grub_env_get ("lang"), val,
 			       0);
-  if (err)
-    grub_print_error ();
-
   return grub_strdup (val);
 }
 
@@ -507,18 +496,14 @@ static grub_command_t cmd;
 GRUB_MOD_INIT (gettext)
 {
   const char *lang;
-  grub_err_t err;
+  grub_err_t __attribute__((__unused__)) err;
 
   lang = grub_env_get ("lang");
 
   err = grub_gettext_init_ext (&main_context, lang, grub_env_get ("locale_dir"),
 			       grub_env_get ("prefix"));
-  if (err)
-    grub_print_error ();
   err = grub_gettext_init_ext (&secondary_context, lang,
 			       grub_env_get ("secondary_locale_dir"), 0);
-  if (err)
-    grub_print_error ();
 
   grub_register_variable_hook ("locale_dir", NULL, read_main);
   grub_register_variable_hook ("secondary_locale_dir", NULL, read_secondary);
diff --git a/grub-core/normal/menu.c b/grub-core/normal/menu.c
index b946c83..676efa4 100644
--- a/grub-core/normal/menu.c
+++ b/grub-core/normal/menu.c
@@ -813,16 +813,18 @@ run_menu (grub_menu_t menu, int nested, int *auto_boot, int *notify_boot)
 
 /* Callback invoked immediately before a menu entry is executed.  */
 static void
-notify_booting (grub_menu_entry_t entry, void *userdata)
+notify_booting (grub_menu_entry_t entry __attribute__((__unused__)), void *userdata)
 {
-  int *notify_boot = userdata;
+  int * __attribute__((__unused__)) notify_boot = userdata;
 
+#if 0
   if (*notify_boot)
     {
       grub_printf ("  ");
       grub_printf_ (N_("Booting `%s'"), entry->title);
       grub_printf ("\n\n");
     }
+#endif
 }
 
 /* Callback invoked when a default menu entry executed because of a timeout
diff --git a/grub-core/normal/menu_entry.c b/grub-core/normal/menu_entry.c
index 8b0d17e..53b5a98 100644
--- a/grub-core/normal/menu_entry.c
+++ b/grub-core/normal/menu_entry.c
@@ -1176,9 +1176,6 @@ run (struct screen *screen)
   char *dummy[1] = { NULL };
 
   grub_cls ();
-  grub_printf ("  ");
-  grub_printf_ (N_("Booting a command list"));
-  grub_printf ("\n\n");
 
   errs_before = grub_err_printed_errors;
 
