diff -up grub-2.12-rc1/grub-core/boot/i386/pc/boot.S.12~ grub-2.12-rc1/grub-core/boot/i386/pc/boot.S
--- grub-2.12-rc1/grub-core/boot/i386/pc/boot.S.12~	2023-07-10 17:43:48.000000000 +0200
+++ grub-2.12-rc1/grub-core/boot/i386/pc/boot.S	2023-10-20 21:40:06.399274336 +0200
@@ -249,9 +249,6 @@ real_start:
 	/* save drive reference first thing! */
 	pushw	%dx
 
-	/* print a notification message on the screen */
-	MSG(notification_string)
-
 	/* set %si to the disk address packet */
 	movw	$disk_address_packet, %si
 
diff -up grub-2.12-rc1/grub-core/boot/i386/pc/diskboot.S.12~ grub-2.12-rc1/grub-core/boot/i386/pc/diskboot.S
--- grub-2.12-rc1/grub-core/boot/i386/pc/diskboot.S.12~	2023-07-10 17:43:48.000000000 +0200
+++ grub-2.12-rc1/grub-core/boot/i386/pc/diskboot.S	2023-10-20 21:40:06.399274336 +0200
@@ -50,11 +50,6 @@ _start:
 	/* save drive reference first thing! */
 	pushw	%dx
 
-	/* print a notification message on the screen */
-	pushw	%si
-	MSG(notification_string)
-	popw	%si
-
 	/* this sets up for the first run through "bootloop" */
 	movw	$LOCAL(firstlist), %di
 
diff -up grub-2.12-rc1/grub-core/gettext/gettext.c.12~ grub-2.12-rc1/grub-core/gettext/gettext.c
--- grub-2.12-rc1/grub-core/gettext/gettext.c.12~	2023-07-10 17:43:48.000000000 +0200
+++ grub-2.12-rc1/grub-core/gettext/gettext.c	2023-10-20 21:41:24.811351600 +0200
@@ -434,16 +434,12 @@ static char *
 grub_gettext_env_write_lang (struct grub_env_var *var
 			     __attribute__ ((unused)), const char *val)
 {
-  grub_err_t err;
+  grub_err_t __attribute__((__unused__)) err;
   err = grub_gettext_init_ext (&main_context, val, grub_env_get ("locale_dir"),
 			       grub_env_get ("prefix"));
-  if (err)
-    grub_print_error ();
 
   err = grub_gettext_init_ext (&secondary_context, val,
 			       grub_env_get ("secondary_locale_dir"), 0);
-  if (err)
-    grub_print_error ();
 
   return grub_strdup (val);
 }
@@ -451,23 +447,19 @@ grub_gettext_env_write_lang (struct grub
 void
 grub_gettext_reread_prefix (const char *val)
 {
-  grub_err_t err;
+  grub_err_t __attribute__((__unused__)) err;
   err = grub_gettext_init_ext (&main_context, grub_env_get ("lang"),
 			       grub_env_get ("locale_dir"),
 			       val);
-  if (err)
-    grub_print_error ();
 }
 
 static char *
 read_main (struct grub_env_var *var
 	   __attribute__ ((unused)), const char *val)
 {
-  grub_err_t err;
+  grub_err_t __attribute__((__unused__)) err;
   err = grub_gettext_init_ext (&main_context, grub_env_get ("lang"), val,
 			       grub_env_get ("prefix"));
-  if (err)
-    grub_print_error ();
   return grub_strdup (val);
 }
 
@@ -475,12 +467,9 @@ static char *
 read_secondary (struct grub_env_var *var
 		__attribute__ ((unused)), const char *val)
 {
-  grub_err_t err;
+  grub_err_t __attribute__((__unused__)) err;
   err = grub_gettext_init_ext (&secondary_context, grub_env_get ("lang"), val,
 			       0);
-  if (err)
-    grub_print_error ();
-
   return grub_strdup (val);
 }
 
@@ -500,18 +489,14 @@ grub_cmd_translate (grub_command_t cmd _
 GRUB_MOD_INIT (gettext)
 {
   const char *lang;
-  grub_err_t err;
+  grub_err_t __attribute__((__unused__)) err;
 
   lang = grub_env_get ("lang");
 
   err = grub_gettext_init_ext (&main_context, lang, grub_env_get ("locale_dir"),
 			       grub_env_get ("prefix"));
-  if (err)
-    grub_print_error ();
   err = grub_gettext_init_ext (&secondary_context, lang,
 			       grub_env_get ("secondary_locale_dir"), 0);
-  if (err)
-    grub_print_error ();
 
   grub_register_variable_hook ("locale_dir", NULL, read_main);
   grub_register_variable_hook ("secondary_locale_dir", NULL, read_secondary);
diff -up grub-2.12-rc1/grub-core/kern/main.c.12~ grub-2.12-rc1/grub-core/kern/main.c
diff -up grub-2.12-rc1/grub-core/normal/menu.c.12~ grub-2.12-rc1/grub-core/normal/menu.c
--- grub-2.12-rc1/grub-core/normal/menu.c.12~	2023-10-20 21:40:06.400274350 +0200
+++ grub-2.12-rc1/grub-core/normal/menu.c	2023-10-20 21:43:11.000745119 +0200
@@ -814,16 +814,18 @@ run_menu (grub_menu_t menu, int nested,
 
 /* Callback invoked immediately before a menu entry is executed.  */
 static void
-notify_booting (grub_menu_entry_t entry, void *userdata)
+notify_booting (grub_menu_entry_t entry __attribute__((__unused__)), void *userdata)
 {
-  int *notify_boot = userdata;
+  int * __attribute__((__unused__)) notify_boot = userdata;
 
+#if 0
   if (*notify_boot)
     {
       grub_printf ("  ");
       grub_printf_ (N_("Booting `%s'"), entry->title);
       grub_printf ("\n\n");
     }
+#endif
 }
 
 /* Callback invoked when a default menu entry executed because of a timeout
diff -up grub-2.12-rc1/grub-core/normal/menu_entry.c.12~ grub-2.12-rc1/grub-core/normal/menu_entry.c
--- grub-2.12-rc1/grub-core/normal/menu_entry.c.12~	2023-07-10 17:43:48.000000000 +0200
+++ grub-2.12-rc1/grub-core/normal/menu_entry.c	2023-10-20 21:40:06.400274350 +0200
@@ -1176,9 +1176,6 @@ run (struct screen *screen)
   char *dummy[1] = { NULL };
 
   grub_cls ();
-  grub_printf ("  ");
-  grub_printf_ (N_("Booting a command list"));
-  grub_printf ("\n\n");
 
   errs_before = grub_err_printed_errors;
 
