diff -Naur grub-2.02-20180928/grub-core/boot/i386/pc/boot.S grub-2.02-20180928-p/grub-core/boot/i386/pc/boot.S
--- grub-2.02-20180928/grub-core/boot/i386/pc/boot.S	2018-09-27 14:56:45.000000000 +0200
+++ grub-2.02-20180928-p/grub-core/boot/i386/pc/boot.S	2019-02-14 01:40:39.323573603 +0100
@@ -249,9 +249,6 @@
 	/* save drive reference first thing! */
 	pushw	%dx
 
-	/* print a notification message on the screen */
-	MSG(notification_string)
-
 	/* set %si to the disk address packet */
 	movw	$disk_address_packet, %si
 
diff -Naur grub-2.02-20180928/grub-core/boot/i386/pc/diskboot.S grub-2.02-20180928-p/grub-core/boot/i386/pc/diskboot.S
--- grub-2.02-20180928/grub-core/boot/i386/pc/diskboot.S	2018-09-27 14:56:45.000000000 +0200
+++ grub-2.02-20180928-p/grub-core/boot/i386/pc/diskboot.S	2019-02-14 01:40:39.323573603 +0100
@@ -50,11 +50,6 @@
 	/* save drive reference first thing! */
 	pushw	%dx
 
-	/* print a notification message on the screen */
-	pushw	%si
-	MSG(notification_string)
-	popw	%si
-
 	/* this sets up for the first run through "bootloop" */
 	movw	$LOCAL(firstlist), %di
 
diff -Naur grub-2.02-20180928/grub-core/gettext/gettext.c grub-2.02-20180928-p/grub-core/gettext/gettext.c
--- grub-2.02-20180928/grub-core/gettext/gettext.c	2018-09-27 14:56:45.000000000 +0200
+++ grub-2.02-20180928-p/grub-core/gettext/gettext.c	2019-02-14 01:40:39.323573603 +0100
@@ -434,16 +434,12 @@
 grub_gettext_env_write_lang (struct grub_env_var *var
 			     __attribute__ ((unused)), const char *val)
 {
-  grub_err_t err;
+  grub_err_t __attribute__((__unused__)) err;
   err = grub_gettext_init_ext (&main_context, val, grub_env_get ("locale_dir"),
 			       grub_env_get ("prefix"));
-  if (err)
-    grub_print_error ();
 
   err = grub_gettext_init_ext (&secondary_context, val,
 			       grub_env_get ("secondary_locale_dir"), 0);
-  if (err)
-    grub_print_error ();
 
   return grub_strdup (val);
 }
@@ -451,23 +447,19 @@
 void
 grub_gettext_reread_prefix (const char *val)
 {
-  grub_err_t err;
+  grub_err_t __attribute__((__unused__)) err;
   err = grub_gettext_init_ext (&main_context, grub_env_get ("lang"), 
 			       grub_env_get ("locale_dir"),
 			       val);
-  if (err)
-    grub_print_error ();
 }
 
 static char *
 read_main (struct grub_env_var *var
 	   __attribute__ ((unused)), const char *val)
 {
-  grub_err_t err;
+  grub_err_t __attribute__((__unused__)) err;
   err = grub_gettext_init_ext (&main_context, grub_env_get ("lang"), val,
 			       grub_env_get ("prefix"));
-  if (err)
-    grub_print_error ();
   return grub_strdup (val);
 }
 
@@ -475,12 +467,9 @@
 read_secondary (struct grub_env_var *var
 		__attribute__ ((unused)), const char *val)
 {
-  grub_err_t err;
+  grub_err_t __attribute__((__unused__)) err;
   err = grub_gettext_init_ext (&secondary_context, grub_env_get ("lang"), val,
 			       0);
-  if (err)
-    grub_print_error ();
-
   return grub_strdup (val);
 }
 
@@ -500,18 +489,14 @@
 GRUB_MOD_INIT (gettext)
 {
   const char *lang;
-  grub_err_t err;
+  grub_err_t __attribute__((__unused__)) err;
 
   lang = grub_env_get ("lang");
 
   err = grub_gettext_init_ext (&main_context, lang, grub_env_get ("locale_dir"),
 			       grub_env_get ("prefix"));
-  if (err)
-    grub_print_error ();
   err = grub_gettext_init_ext (&secondary_context, lang,
 			       grub_env_get ("secondary_locale_dir"), 0);
-  if (err)
-    grub_print_error ();
 
   grub_register_variable_hook ("locale_dir", NULL, read_main);
   grub_register_variable_hook ("secondary_locale_dir", NULL, read_secondary);
diff -Naur grub-2.02-20180928/grub-core/kern/main.c grub-2.02-20180928-p/grub-core/kern/main.c
--- grub-2.02-20180928/grub-core/kern/main.c	2018-09-27 14:56:45.000000000 +0200
+++ grub-2.02-20180928-p/grub-core/kern/main.c	2019-02-14 01:40:39.323573603 +0100
@@ -269,11 +269,6 @@
 
   grub_boot_time ("After machine init.");
 
-  /* Hello.  */
-  grub_setcolorstate (GRUB_TERM_COLOR_HIGHLIGHT);
-  grub_printf ("Welcome to GRUB!\n\n");
-  grub_setcolorstate (GRUB_TERM_COLOR_STANDARD);
-
   grub_load_config ();
 
   grub_boot_time ("Before loading embedded modules.");
