--- grub-2.02/util/grub.d/30_os-prober.in~	2017-04-24 10:15:50.021585392 +0100
+++ grub-2.02/util/grub.d/30_os-prober.in	2017-12-10 22:24:27.253415882 +0000
@@ -207,7 +207,7 @@
         LBOOT="`echo ${LINUX} | cut -d ':' -f 2`"
         LLABEL="`echo ${LINUX} | cut -d ':' -f 3 | tr '^' ' '`"
         LKERNEL="`echo ${LINUX} | cut -d ':' -f 4`"
-        LINITRD="`echo ${LINUX} | cut -d ':' -f 5`"
+        LINITRD="`echo ${LINUX} | cut -d ':' -f 5 | tr '^' ' '`"
         LPARAMS="`echo ${LINUX} | cut -d ':' -f 6- | tr '^' ' '`"
 
         if [ -z "${LLABEL}" ] ; then
