diff --git a/grub-core/kern/misc.c b/grub-core/kern/misc.c
index 18c34b8..96bcfcf 100644
--- a/grub-core/kern/misc.c
+++ b/grub-core/kern/misc.c
@@ -1432,6 +1432,15 @@ grub_abort (void)
   grub_exit ();
 }
 
+#if defined(__aarch64__) && !defined (GRUB_UTIL)
+/* On aarch64, we get references to abort().  */
+void __attribute__ ((noreturn)) __attribute__((visibility("default")))
+abort (void)
+{
+  grub_abort ();
+}
+#endif
+
 void
 grub_fatal (const char *fmt, ...)
 {
