diff -up grub-2.06/grub-core/gfxmenu/gui_label.c.1~ grub-2.06/grub-core/gfxmenu/gui_label.c
--- grub-2.06/grub-core/gfxmenu/gui_label.c.1~	2021-03-02 15:54:20.000000000 +0100
+++ grub-2.06/grub-core/gfxmenu/gui_label.c	2022-06-26 18:03:04.430634087 +0200
@@ -25,6 +25,11 @@
 #include <grub/i18n.h>
 #include <grub/color.h>
 
+#define LABEL_ENTER "Boot entry     "
+#define LABEL_E     "Edit entry     "
+#define LABEL_C     "Console     "
+#define LABEL_ESC   "Back"
+
 static const char *align_options[] =
 {
   "left",
@@ -192,6 +197,15 @@ label_set_property (void *vself, const c
 	       "or `c' for a command-line.");
 	   else if (grub_strcmp (value, "@KEYMAP_SHORT@") == 0)
 	    value = _("enter: boot, `e': options, `c': cmd-line");
+
+	   else if (grub_strcmp (value, LABEL_ENTER) == 0)
+	    value = _(LABEL_ENTER);
+	   else if (grub_strcmp (value, LABEL_E) == 0)
+	    value = _(LABEL_E);
+	   else if (grub_strcmp (value, LABEL_C) == 0)
+	    value = _(LABEL_C);
+	   else if (grub_strcmp (value, LABEL_ESC) == 0)
+	    value = _(LABEL_ESC);
 	   /* FIXME: Add more templates here if needed.  */
 
 	  if (grub_printf_fmt_check(value, "%d") != GRUB_ERR_NONE)
