diff --git a/configure.ac b/configure.ac
index 47a96c3..23f7886 100644
--- a/configure.ac
+++ b/configure.ac
@@ -978,7 +978,7 @@ if test x"$platform" != xemu ; then
        AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]], [[]])],
 		         [grub_cv_target_cc_soft_float="-m4-nofpu"], [])
     fi
-    for cand in "-msoft-float -Xclang -msoft-float -Xclang -no-implicit-float" \
+    for cand in "-msoft-float -Xclang -no-implicit-float" \
 		"-Xclang -msoft-float -Xclang -no-implicit-float" \
 		"-Xclang -msoft-float" "-msoft-float"; do
       if test x"$grub_cv_target_cc_soft_float" != xno ; then
