diff -up grub-2.02-20180928/grub-core/kern/misc.c.omv~ grub-2.02-20180928/grub-core/kern/misc.c
--- grub-2.02-20180928/grub-core/kern/misc.c.omv~	2018-12-23 22:58:51.329679479 +0100
+++ grub-2.02-20180928/grub-core/kern/misc.c	2018-12-23 22:59:00.087803621 +0100
@@ -1098,6 +1098,15 @@ grub_abort (void)
   grub_exit ();
 }
 
+#if defined (__clang__) && !defined (GRUB_UTIL)
+/* clang emits references to abort().  */
+void __attribute__ ((noreturn))
+abort (void)
+{
+  grub_abort ();
+}
+#endif
+
 void
 grub_fatal (const char *fmt, ...)
 {
diff -up grub-2.02-20180928/include/grub/misc.h.omv~ grub-2.02-20180928/include/grub/misc.h
