diff -up grub-2.12-rc1/Makefile.am.19~ grub-2.12-rc1/Makefile.am
--- grub-2.12-rc1/Makefile.am.19~	2023-10-20 21:45:00.431116079 +0200
+++ grub-2.12-rc1/Makefile.am	2023-10-20 21:45:00.466116508 +0200
@@ -434,7 +434,7 @@ if COND_i386_coreboot
 FS_PAYLOAD_MODULES ?= $(shell cat grub-core/fs.lst)
 default_payload.elf: grub-mkstandalone grub-mkimage FORCE
 	test -f $@ && rm $@ || true
-	pkgdatadir=. ./grub-mkstandalone --grub-mkimage=./grub-mkimage -O i386-coreboot -o $@ --modules='ahci pata ehci uhci ohci usb_keyboard usbms part_msdos ext2 fat at_keyboard part_gpt usbserial_usbdebug cbfs' --install-modules='ls linux search configfile normal cbtime cbls memrw iorw minicmd lsmmap lspci halt reboot hexdump pcidump regexp setpci lsacpi chain test serial multiboot cbmemc linux16 gzio echo help syslinuxcfg xnu $(FS_PAYLOAD_MODULES) password_pbkdf2 $(EXTRA_PAYLOAD_MODULES)' --fonts= --themes= --locales= -d grub-core/ /boot/grub/grub.cfg=$(srcdir)/coreboot.cfg
+	pkgdatadir=. ./grub-mkstandalone --grub-mkimage=./grub-mkimage -O i386-coreboot -o $@ --modules='ahci pata ehci uhci ohci usb_keyboard usbms part_msdos ext2 fat at_keyboard part_gpt usbserial_usbdebug cbfs' --install-modules='ls linux search configfile normal cbtime cbls memrw iorw minicmd lsmmap lspci halt reboot hexdump pcidump regexp setpci lsacpi chain test serial multiboot cbmemc linux16 gzio echo help syslinuxcfg xnu $(FS_PAYLOAD_MODULES) password_pbkdf2 $(EXTRA_PAYLOAD_MODULES)' --fonts= --themes= --locales= -d grub-core/ /boot/grub2/grub.cfg=$(srcdir)/coreboot.cfg
 endif
 
 endif
diff -up grub-2.12-rc1/util/grub-install.c.19~ grub-2.12-rc1/util/grub-install.c
--- grub-2.12-rc1/util/grub-install.c.19~	2023-07-10 17:43:48.000000000 +0200
+++ grub-2.12-rc1/util/grub-install.c	2023-10-20 21:45:00.466116508 +0200
@@ -1694,7 +1694,7 @@ main (int argc, char *argv[])
     case GRUB_INSTALL_PLATFORM_I386_IEEE1275:
     case GRUB_INSTALL_PLATFORM_POWERPC_IEEE1275:
       {
-	char *dst = grub_util_path_concat (2, grubdir, "grub");
+	char *dst = grub_util_path_concat (2, grubdir, "grub2");
 	grub_install_copy_file (imgfile, dst, 1);
 	free (dst);
       }
diff -up grub-2.12-rc1/util/grub-mkimage.c.19~ grub-2.12-rc1/util/grub-mkimage.c
--- grub-2.12-rc1/util/grub-mkimage.c.19~	2023-07-10 17:43:48.000000000 +0200
+++ grub-2.12-rc1/util/grub-mkimage.c	2023-10-20 21:45:00.467116521 +0200
@@ -69,7 +69,7 @@ static struct argp_option options[] = {
    /* TRANSLATORS: "memdisk" here isn't an identifier, it can be translated.
     "embed" is a verb (command description).  "*/
    N_("embed FILE as a memdisk image\n"
-      "Implies `-p (memdisk)/boot/grub' and overrides any prefix supplied previously,"
+      "Implies `-p (memdisk)/boot/grub2' and overrides any prefix supplied previously,"
       " but the prefix itself can be overridden by later options"), 0},
   {"dtb",  'D', N_("FILE"), 0, N_("embed FILE as a device tree (DTB)\n"), 0},
    /* TRANSLATORS: "embed" is a verb (command description).  "*/
@@ -179,7 +179,7 @@ argp_parser (int key, char *arg, struct
       if (arguments->prefix)
 	free (arguments->prefix);
 
-      arguments->prefix = xstrdup ("(memdisk)/boot/grub");
+      arguments->prefix = xstrdup ("(memdisk)/boot/grub2");
       break;
 
     case 'D':
diff -up grub-2.12-rc1/util/grub-mkrescue.c.19~ grub-2.12-rc1/util/grub-mkrescue.c
--- grub-2.12-rc1/util/grub-mkrescue.c.19~	2023-07-10 17:43:48.000000000 +0200
+++ grub-2.12-rc1/util/grub-mkrescue.c	2023-10-20 21:46:57.258521023 +0200
@@ -281,14 +281,14 @@ make_image_abs (enum grub_install_plat p
 	fprintf (load_cfg_f, "search --set=root --fs-uuid %s\n", iso_uuid);
 	break;
     }
-  fprintf (load_cfg_f, "set prefix=(${root})/boot/grub\n");
+  fprintf (load_cfg_f, "set prefix=(${root})/boot/grub2\n");
 
   write_part (load_cfg_f, source_dirs[plat]);
   fclose (load_cfg_f);
 
   grub_install_push_module ("search");
   grub_install_push_module ("iso9660");
-  grub_install_make_image_wrap (source_dirs[plat], "/boot/grub", output,
+  grub_install_make_image_wrap (source_dirs[plat], "/boot/grub2", output,
 				0, load_cfg,
 				mkimage_target, 0);
   grub_install_pop_module ();
@@ -327,7 +327,7 @@ make_image_fwdisk_abs (enum grub_install
   fclose (load_cfg_f);
 
   grub_install_push_module ("iso9660");
-  grub_install_make_image_wrap (source_dirs[plat], "()/boot/grub", output,
+  grub_install_make_image_wrap (source_dirs[plat], "()/boot/grub2", output,
 				0, load_cfg, mkimage_target, 0);
   grub_install_pop_module ();
   grub_util_unlink (load_cfg);
@@ -517,7 +517,7 @@ main (int argc, char *argv[])
 
   iso9660_dir = grub_util_make_temporary_dir ();
   grub_util_info ("temporary iso9660 dir is `%s'", iso9660_dir);
-  boot_grub = grub_util_path_concat (3, iso9660_dir, "boot", "grub");
+  boot_grub = grub_util_path_concat (3, iso9660_dir, "boot", "grub2");
   grub_install_mkdir_p (boot_grub);
   romdir = grub_util_path_concat (2, boot_grub, "roms");
   grub_util_mkdir (romdir);
@@ -617,12 +617,12 @@ main (int argc, char *argv[])
       grub_install_push_module ("biosdisk");
       grub_install_push_module ("iso9660");
       grub_install_make_image_wrap (source_dirs[GRUB_INSTALL_PLATFORM_I386_PC],
-				    "/boot/grub", output,
+				    "/boot/grub2", output,
 				    0, load_cfg,
 				    "i386-pc-eltorito", 0);
 
       xorriso_push ("-b");
-      xorriso_push ("boot/grub/i386-pc/eltorito.img");
+      xorriso_push ("boot/grub2/i386-pc/eltorito.img");
       xorriso_push ("-no-emul-boot");
       xorriso_push ("-boot-load-size");
       xorriso_push ("4");
@@ -661,7 +661,7 @@ main (int argc, char *argv[])
 	      fwrite (buf, 1, 512, sa);
 
 	      grub_install_make_image_wrap_file (source_dirs[GRUB_INSTALL_PLATFORM_I386_PC],
-						 "/boot/grub", sa, sysarea_img,
+						 "/boot/grub2", sa, sysarea_img,
 						 0, load_cfg,
 						 "i386-pc", 0);
 	      sz = ftello (sa);
@@ -888,8 +888,8 @@ main (int argc, char *argv[])
       grub_install_copy_file (grub_chrp, bootx, 1);
       grub_install_mkdir_p (ppc_chrp);
       grub_install_copy_file (bisrc, bitgt, 1);
-      xorriso_link ("/System/Library/CoreServices/grub.elf", "/boot/grub/powerpc-ieee1275/core.elf");
-      xorriso_link ("/boot/grub/powerpc.elf", "/boot/grub/powerpc-ieee1275/core.elf");
+      xorriso_link ("/System/Library/CoreServices/grub.elf", "/boot/grub2/powerpc-ieee1275/core.elf");
+      xorriso_link ("/boot/grub2/powerpc.elf", "/boot/grub2/powerpc-ieee1275/core.elf");
       /* FIXME: add PreP */
       if (system_area == SYS_AREA_COMMON)
 	{
@@ -938,25 +938,25 @@ main (int argc, char *argv[])
       xorriso_push ("-B");
       xorriso_push (",");
       xorriso_push ("--grub2-sparc-core");
-      xorriso_push ("/boot/grub/sparc64-ieee1275/core.img");
+      xorriso_push ("/boot/grub2/sparc64-ieee1275/core.img");
     }
 
   make_image_fwdisk (GRUB_INSTALL_PLATFORM_MIPS_ARC, "mips-arc", "mips-arc/core.img");
 
   if (source_dirs[GRUB_INSTALL_PLATFORM_MIPS_ARC])
     {
-      xorriso_link ("/boot/grub/mips-arc/grub", "/boot/grub/mips-arc/core.img");
-      xorriso_link ("/boot/grub/mips-arc/sashARCS", "/boot/grub/mips-arc/core.img");
-      xorriso_link ("/boot/grub/mips-arc/sash", "/boot/grub/mips-arc/core.img");
+      xorriso_link ("/boot/grub2/mips-arc/grub", "/boot/grub2/mips-arc/core.img");
+      xorriso_link ("/boot/grub2/mips-arc/sashARCS", "/boot/grub2/mips-arc/core.img");
+      xorriso_link ("/boot/grub2/mips-arc/sash", "/boot/grub2/mips-arc/core.img");
     }
   if (source_dirs[GRUB_INSTALL_PLATFORM_MIPS_ARC] && system_area == SYS_AREA_ARCS)
     {
       xorriso_push ("-mips-boot");
-      xorriso_push ("/boot/grub/mips-arc/sashARCS");
+      xorriso_push ("/boot/grub2/mips-arc/sashARCS");
       xorriso_push ("-mips-boot");
-      xorriso_push ("/boot/grub/mips-arc/sash");
+      xorriso_push ("/boot/grub2/mips-arc/sash");
       xorriso_push ("-mips-boot");
-      xorriso_push ("/boot/grub/mips-arc/grub");
+      xorriso_push ("/boot/grub2/mips-arc/grub");
     }
 
   make_image_fwdisk (GRUB_INSTALL_PLATFORM_MIPSEL_ARC, "mipsel-arc", "arc.exe");
diff -up grub-2.12-rc1/util/grub-mkstandalone.c.19~ grub-2.12-rc1/util/grub-mkstandalone.c
--- grub-2.12-rc1/util/grub-mkstandalone.c.19~	2023-07-10 17:43:48.000000000 +0200
+++ grub-2.12-rc1/util/grub-mkstandalone.c	2023-10-20 21:47:30.910916237 +0200
@@ -112,7 +112,7 @@ argp_parser (int key, char *arg, struct
 
 struct argp argp = {
   options, argp_parser, N_("[OPTION] SOURCE..."),
-  N_("Generate a standalone image (containing all modules) in the selected format")"\v"N_("Graft point syntax (E.g. /boot/grub/grub.cfg=./grub.cfg) is accepted"),
+  N_("Generate a standalone image (containing all modules) in the selected format")"\v"N_("Graft point syntax (E.g. /boot/grub2/grub.cfg=./grub.cfg) is accepted"),
   NULL, help_filter, NULL
 };
 
@@ -314,7 +314,7 @@ main (int argc, char *argv[])
   enum grub_install_plat plat = grub_install_get_target (grub_install_source_directory);
 
   char *memdisk_dir = grub_util_make_temporary_dir ();
-  char *boot_grub = grub_util_path_concat (3, memdisk_dir, "boot", "grub");
+  char *boot_grub = grub_util_path_concat (3, memdisk_dir, "boot", "grub2");
   grub_install_copy_files (grub_install_source_directory,
 			   boot_grub, plat);
 
@@ -354,7 +354,7 @@ main (int argc, char *argv[])
   grub_install_push_module ("tar");
 
   grub_install_make_image_wrap (grub_install_source_directory,
-				"(memdisk)/boot/grub", output_image,
+				"(memdisk)/boot/grub2", output_image,
 				memdisk_img, NULL,
 				grub_util_get_target_name (format), 0);
 
