--- grub-2.00/util/grub.d/30_os-prober.in.0504~	2013-07-04 16:09:32.437050565 +0200
+++ grub-2.00/util/grub.d/30_os-prober.in	2013-07-04 16:12:26.444568209 +0200
@@ -41,6 +41,15 @@ if [ -z "${OSPROBED}" ] ; then
   exit 0
 fi
 
+# If secure boot is enabled, use linuxefi/initrdefi for Linux (bnc#810912)
+if [ -d /sys/firmware/efi ] && [ "x${GRUB_USE_LINUXEFI}" = "xtrue" ]; then
+  LINUX_LOADER_CMD=linuxefi
+  LINUX_INITRD_CMD=initrdefi
+else
+  LINUX_LOADER_CMD=linux
+  LINUX_INITRD_CMD=initrd
+fi
+
 osx_entry() {
     if [ x$2 = x32 ]; then
         # TRANSLATORS: it refers to kernel architecture (32-bit)
@@ -200,11 +209,11 @@ EOF
 	    save_default_entry | grub_add_tab
 	    printf '%s\n' "${prepare_boot_cache}"
 	    cat <<  EOF
-	linux ${LKERNEL} ${LPARAMS}
+	${LINUX_LOADER_CMD} ${LKERNEL} ${LPARAMS}
 EOF
             if [ -n "${LINITRD}" ] ; then
           cat << EOF
-	initrd ${LINITRD}
+	${LINUX_INITRD_CMD} ${LINITRD}
 EOF
             fi
         cat << EOF
@@ -220,11 +229,11 @@ EOF
 	save_default_entry | sed -e "s/^/$grub_tab$grub_tab/"
 	printf '%s\n' "${prepare_boot_cache}" | grub_add_tab
 	cat <<  EOF
-		linux ${LKERNEL} ${LPARAMS}
+		${LINUX_LOADER_CMD} ${LKERNEL} ${LPARAMS}
 EOF
         if [ -n "${LINITRD}" ] ; then
             cat << EOF
-		initrd ${LINITRD}
+		${LINUX_INITRD_CMD} ${LINITRD}
 EOF
         fi
         cat << EOF
