--- grub-2.00/util/grub.d/30_os-prober.in.0502~	2013-07-04 15:52:04.756231698 +0200
+++ grub-2.00/util/grub.d/30_os-prober.in	2013-07-04 15:56:14.524784982 +0200
@@ -127,6 +127,11 @@ for OS in ${OSPROBED} ; do
 	BTRFSsubvol="`echo ${OS} | cut -d ':' -f 7`"
   fi
 
+  CLASS="$(echo ${LABEL} | tr '[A-Z]' '[a-z]' | sed 's,[0-9]*$,,')" 
+  if [ -z "${CLASS}" ] ; then
+    CLASS="unknown";
+  fi
+
   if [ -z "${LONGNAME}" ] ; then
     LONGNAME="${LABEL}"
   fi
@@ -220,7 +225,7 @@ EOF
 
 	if [ "x$is_first_entry" = xtrue ]; then
             cat << EOF
-menuentry '$(echo "$OS $onstr" | grub_quote)' --class gnu-linux --class gnu --class os \$menuentry_id_option 'osprober-gnulinux-simple-$boot_device_id' {
+menuentry '$(echo "$OS $onstr" | grub_quote)' --class ${CLASS} --class gnu-linux --class gnu --class os \$menuentry_id_option 'osprober-gnulinux-simple-$boot_device_id' {
 EOF
 	    save_default_entry | grub_add_tab
 	    printf '%s\n' "${prepare_boot_cache}"
@@ -240,7 +245,7 @@ EOF
 	fi
 	title="${LLABEL} $onstr"
         cat << EOF
-	menuentry '$(echo "$title" | grub_quote)' --class gnu-linux --class gnu --class os \$menuentry_id_option 'osprober-gnulinux-$LKERNEL-${recovery_params}-$boot_device_id' {
+	menuentry '$(echo "$title" | grub_quote)' --class ${CLASS} --class gnu-linux --class gnu --class os \$menuentry_id_option 'osprober-gnulinux-$LKERNEL-${recovery_params}-$boot_device_id' {
 EOF
 	save_default_entry | sed -e "s/^/$grub_tab$grub_tab/"
 	printf '%s\n' "${prepare_boot_cache}" | grub_add_tab
