diff -Naur grub-2.04/Makefile.am grub-2.04.tpg/Makefile.am
--- grub-2.04/Makefile.am	2019-10-24 08:49:04.756376834 +0000
+++ grub-2.04.tpg/Makefile.am	2019-10-24 08:50:49.812385345 +0000
@@ -5,7 +5,7 @@
 if COND_real_platform
 SUBDIRS += grub-core
 endif
-SUBDIRS += po docs util/bash-completion.d
+SUBDIRS += po util/bash-completion.d
 
 include $(top_srcdir)/conf/Makefile.common
 include $(top_srcdir)/conf/Makefile.extra-dist
diff -Naur grub-2.04/configure.ac grub-2.04.tpg/configure.ac
--- grub-2.04/configure.ac	2019-10-24 08:49:04.740376832 +0000
+++ grub-2.04.tpg/configure.ac	2019-10-24 08:51:16.409387500 +0000
@@ -2017,7 +2017,6 @@
 AC_CONFIG_FILES([grub-core/Makefile])
 AC_CONFIG_FILES([grub-core/lib/gnulib/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
-AC_CONFIG_FILES([docs/Makefile])
 AC_CONFIG_FILES([util/bash-completion.d/Makefile])
 AC_CONFIG_FILES([stamp-h], [echo timestamp > stamp-h])
 AC_CONFIG_FILES([config.h])
