--- grub-2.00/include/grub/term.h.0489~	2013-07-04 15:36:29.492843842 +0200
+++ grub-2.00/include/grub/term.h	2013-07-04 15:37:06.723378965 +0200
@@ -212,8 +212,8 @@ struct grub_term_output
 };
 typedef struct grub_term_output *grub_term_output_t;
 
-#define GRUB_TERM_DEFAULT_NORMAL_COLOR 0x07
-#define GRUB_TERM_DEFAULT_HIGHLIGHT_COLOR 0x70
+#define GRUB_TERM_DEFAULT_NORMAL_COLOR 0x0F
+#define GRUB_TERM_DEFAULT_HIGHLIGHT_COLOR 0xF0
 #define GRUB_TERM_DEFAULT_STANDARD_COLOR 0x07
 
 /* Current color state.  */
