diff --git a/grub-core/disk/diskfilter.c b/grub-core/disk/diskfilter.c
index 3a26de6..427be28 100644
--- a/grub-core/disk/diskfilter.c
+++ b/grub-core/disk/diskfilter.c
@@ -951,8 +951,9 @@ grub_diskfilter_write (grub_disk_t disk __attribute ((unused)),
 		 grub_size_t size __attribute ((unused)),
 		 const char *buf __attribute ((unused)))
 {
-  return grub_error (GRUB_ERR_NOT_IMPLEMENTED_YET,
-		     "diskfilter writes are not supported");
+/*  return grub_error (GRUB_ERR_NOT_IMPLEMENTED_YET,
+		     "diskfilter writes are not supported");*/
+    return GRUB_ERR_NONE;
 }
 
 struct grub_diskfilter_vg *
