--- grub-2.00/configure.ac~	2012-12-21 10:31:28.227181112 +0100
+++ grub-2.00/configure.ac	2012-12-21 10:31:31.074226776 +0100
@@ -791,7 +791,8 @@ if test "$platform" = emu; then
 [# Check for curses libraries.]
   AC_CHECK_LIB([ncurses], [wgetch], [LIBCURSES="-lncurses"],
     [AC_CHECK_LIB([curses], [wgetch], [LIBCURSES="-lcurses"],
-      [missing_ncurses=[true]])])
+      [AC_CHECK_LIB([ncursesw], [wgetch], [LIBCURSES="-lncursesw"],
+        [missing_ncurses=[true]])])])
   AC_SUBST([LIBCURSES])
 [if [ x"$missing_ncurses" = x ]; then ]
   [# Check for headers.]
