diff --git a/Makefile.am b/Makefile.am
index 43635d5..e788f6d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -5,7 +5,7 @@ SUBDIRS = grub-core/lib/gnulib .
 if COND_real_platform
 SUBDIRS += grub-core
 endif
-SUBDIRS += po docs util/bash-completion.d
+SUBDIRS += po util/bash-completion.d
 
 include $(top_srcdir)/conf/Makefile.common
 include $(top_srcdir)/conf/Makefile.extra-dist
diff --git a/configure.ac b/configure.ac
index 23f7886..3403316 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2610,7 +2610,6 @@ AC_CONFIG_FILES([grub-core/sbat.csv], [
 ])
 AC_CONFIG_FILES([grub-core/lib/gnulib/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
-AC_CONFIG_FILES([docs/Makefile])
 AC_CONFIG_FILES([util/bash-completion.d/Makefile])
 AC_CONFIG_FILES([stamp-h], [echo timestamp > stamp-h])
 AC_CONFIG_FILES([config.h])
