--- grub-2.02~beta2/util/grub.d/30_os-prober.in.1116~	2014-04-08 15:19:53.701855768 +0200
+++ grub-2.02~beta2/util/grub.d/30_os-prober.in	2014-04-08 15:24:57.322651512 +0200
@@ -139,6 +139,11 @@ for OS in ${OSPROBED} ; do
 	BTRFSsubvol="`echo ${OS} | cut -d ':' -f 7`"
   fi
 
+  CLASS="$(echo ${LABEL} | tr '[A-Z]' '[a-z]' | sed 's,[0-9]*$,,')" 
+  if [ -z "${CLASS}" ] ; then
+    CLASS="unknown";
+  fi
+
   if [ -z "${LONGNAME}" ] ; then
     LONGNAME="${LABEL}"
   fi

