diff --git a/grub-core/osdep/unix/platform.c b/grub-core/osdep/unix/platform.c
index ca448bc..2899c50 100644
--- a/grub-core/osdep/unix/platform.c
+++ b/grub-core/osdep/unix/platform.c
@@ -149,10 +149,12 @@ grub_install_register_efi (grub_device_t efidir_grub_dev,
       grub_util_error (_("%s: not found"), "efibootmgr");
     }
 
-  /* On Linux, we need the efivars kernel modules.  */
+  /* Openmandriva uses the latest kernel interface so the efi
+   * module is not required */
+  /* On Linux, we need the efivars kernel modules.
 #ifdef __linux__
   grub_util_exec ((const char * []){ "modprobe", "-q", "efivars", NULL });
-#endif
+#endif */
   /* Delete old entries from the same distributor.  */
   ret = grub_install_remove_efi_entries_by_distributor (efi_distributor);
   if (ret)
