diff -Naur grub-2.02-20150306/util/grub.d/10_linux.in grub-2.02-20150306.tpg/util/grub.d/10_linux.in
--- grub-2.02-20150306/util/grub.d/10_linux.in	2015-03-06 23:58:02.000000000 +0000
+++ grub-2.02-20150306.tpg/util/grub.d/10_linux.in	2015-03-07 00:05:19.058570172 +0000
@@ -74,6 +74,12 @@
   type="$3"
   args="$4"
 
+  # (tpg) check for swap
+  if swapon -s | grep '/dev/sd.[0-9]\{1,\}'; then
+    swap=`swapon -s | grep '/dev/sd.[0-9]\{1,\}'|awk '{print $1}'`
+    resumepoint="resume=$swap"
+  fi
+
   if [ -z "$boot_device_id" ]; then
       boot_device_id="$(grub_get_device_id "${GRUB_DEVICE}")"
   fi
@@ -129,7 +135,7 @@
   message="$(gettext_printf "Loading Linux %s ..." ${version})"
   sed "s/^/$submenu_indentation/" << EOF
 	echo	'$(echo "$message" | grub_quote)'
-	linux	${rel_dirname}/${basename} root=${linux_root_device_thisversion} ro ${args}
+	linux	${rel_dirname}/${basename} root=${linux_root_device_thisversion} ro ${args} ${resumepoint}
 EOF
   if test -n "${initrd}" ; then
     # TRANSLATORS: ramdisk isn't identifier. Should be translated.
