diff -up grub-2.06/configure.ac.6~ grub-2.06/configure.ac
--- grub-2.06/configure.ac.6~	2022-06-26 18:03:19.903805709 +0200
+++ grub-2.06/configure.ac	2022-06-26 18:03:47.360184038 +0200
@@ -868,7 +868,7 @@ if test x"$platform" != xemu ; then
        AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]], [[]])],
 		         [grub_cv_target_cc_soft_float="-m4-nofpu"], [])
     fi
-    for cand in "-msoft-float -Xclang -msoft-float -Xclang -no-implicit-float" \
+    for cand in "-msoft-float -Xclang -no-implicit-float" \
 		"-Xclang -msoft-float -Xclang -no-implicit-float" \
 		"-Xclang -msoft-float" "-msoft-float"; do
       if test x"$grub_cv_target_cc_soft_float" != xno ; then
