diff -up grub-2.14/util/grub-mkconfig.in.12~ grub-2.14/util/grub-mkconfig.in
--- grub-2.14/util/grub-mkconfig.in.12~	2026-02-07 01:45:32.568119610 +0100
+++ grub-2.14/util/grub-mkconfig.in	2026-02-07 01:47:08.242166529 +0100
@@ -171,6 +171,12 @@ if [ "x${GRUB_DISABLE_UUID}" = "xtrue" ]
   fi
 fi
 
+if [ "x${GRUB_DISABLE_VENDOR_CONF}" = "x" ] || [ "x${GRUB_DISABLE_VENDOR_CONF}" = "xfalse" ]; then
+  if test -e ${sysconfdir}/default/grub.vendor ; then
+    . ${sysconfdir}/default/grub.vendor
+  fi
+fi
+
 # XXX: should this be deprecated at some point?
 if [ "x${GRUB_TERMINAL}" != "x" ] ; then
   GRUB_TERMINAL_INPUT="${GRUB_TERMINAL}"
@@ -256,6 +262,7 @@ export GRUB_DEFAULT \
   GRUB_ENABLE_CRYPTODISK \
   GRUB_BADRAM \
   GRUB_CMDLINE_LINUX_RECOVERY \
+  GRUB_DISABLE_VENDOR_CONF \
   GRUB_OS_PROBER_SKIP_LIST \
   GRUB_DISABLE_SUBMENU \
   GRUB_FORCE_EFI_ALL_VIDEO
