diff --git a/grub-core/commands/loadenv.c b/grub-core/commands/loadenv.c
index 1664458..6983624 100644
--- a/grub-core/commands/loadenv.c
+++ b/grub-core/commands/loadenv.c
@@ -286,7 +286,8 @@ check_blocklists (grub_envblk_t envblk, struct blocklist *blocklists,
   if (total_length != grub_file_size (file))
     {
       /* Maybe sparse, unallocated sectors. No way in GRUB.  */
-      return grub_error (GRUB_ERR_BAD_FILE_TYPE, "sparse file not allowed");
+      /*return grub_error (GRUB_ERR_BAD_FILE_TYPE, "sparse file not allowed")*/
+       return GRUB_ERR_NONE;
     }
 
   /* One more sanity check. Re-read all sectors by blocklists, and compare
