diff --git a/include/grub/term.h b/include/grub/term.h
index 7f1a14c..80e4983 100644
--- a/include/grub/term.h
+++ b/include/grub/term.h
@@ -230,8 +230,8 @@ struct grub_term_output
 };
 typedef struct grub_term_output *grub_term_output_t;
 
-#define GRUB_TERM_DEFAULT_NORMAL_COLOR 0x07
-#define GRUB_TERM_DEFAULT_HIGHLIGHT_COLOR 0x70
+#define GRUB_TERM_DEFAULT_NORMAL_COLOR 0x0F
+#define GRUB_TERM_DEFAULT_HIGHLIGHT_COLOR 0xF0
 #define GRUB_TERM_DEFAULT_STANDARD_COLOR 0x07
 
 /* Current color state.  */
