--- grub-2.00/grub-core/gettext/gettext.c.0495~	2013-07-04 15:24:20.719425520 +0200
+++ grub-2.00/grub-core/gettext/gettext.c	2013-07-04 15:34:34.134187844 +0200
@@ -423,6 +423,14 @@ grub_gettext_init_ext (struct grub_gette
 
       grub_free (lang);
     }
+
+  /* If no translations are available, fall back to untranslated text. */
+  if (err == GRUB_ERR_FILE_NOT_FOUND)
+    {
+      grub_errno = GRUB_ERR_NONE;
+      return 0;
+    }
+
   return err;
 }
 
