--- grub-2.02~beta2/util/grub-mkconfig.in.1112~	2014-04-08 15:17:01.783892795 +0200
+++ grub-2.02~beta2/util/grub-mkconfig.in	2014-04-08 15:19:14.690681077 +0200
@@ -128,18 +128,41 @@ else
     exit 1
 fi
 
-# Device containing our userland.  Typically used for root= parameter.
-GRUB_DEVICE="`${grub_probe} --target=device /`"
-GRUB_DEVICE_UUID="`${grub_probe} --device ${GRUB_DEVICE} --target=fs_uuid 2> /dev/null`" || true
+probe_nfsroot_device () {
+    while read line ; do
+        set -- $line
+        path=$5
+        fstype=$8
+        device=$9
+
+        if [ "x${path}" = "x/" ] &&
+           [ "x${fstype}" = "xnfs" -o "x${fstype}" = "xnfs4" ] ; then
+            echo "$device"
+            return
+        fi
+    done
+} </proc/self/mountinfo
+
+NFSROOT_DEVICE="`probe_nfsroot_device`"
+
+if [ "x${NFSROOT_DEVICE}" != "x" ]; then
+    GRUB_DEVICE="$NFSROOT_DEVICE"
+    GRUB_DEVICE_UUID=""
+    GRUB_FS="unknown"
+else
+    # Device containing our userland.  Typically used for root= parameter.
+    GRUB_DEVICE="`${grub_probe} --target=device /`"
+    GRUB_DEVICE_UUID="`${grub_probe} --device ${GRUB_DEVICE} --target=fs_uuid 2> /dev/null`" || true
+
+    # Filesystem for the device containing our userland.  Used for stuff like
+    # choosing Hurd filesystem module.
+    GRUB_FS="`${grub_probe} --device ${GRUB_DEVICE} --target=fs 2> /dev/null || echo unknown`"
+fi
 
 # Device containing our /boot partition.  Usually the same as GRUB_DEVICE.
 GRUB_DEVICE_BOOT="`${grub_probe} --target=device /boot`"
 GRUB_DEVICE_BOOT_UUID="`${grub_probe} --device ${GRUB_DEVICE_BOOT} --target=fs_uuid 2> /dev/null`" || true
 
-# Filesystem for the device containing our userland.  Used for stuff like
-# choosing Hurd filesystem module.
-GRUB_FS="`${grub_probe} --device ${GRUB_DEVICE} --target=fs 2> /dev/null || echo unknown`"
-
 if [ x"$GRUB_FS" = xunknown ]; then
     GRUB_FS="$(stat -f --printf=%T / || echo unknown)"
 fi
