diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -29,9 +29,15 @@
 CLASS="--class gnu-linux --class gnu --class os"
 
 if [ "x${GRUB_DISTRIBUTOR}" = "x" ] ; then
-  OS=GNU/Linux
+  if [ -e /etc/system-release ]; then
+    OS="$(sed 's, release .*$,,g' /etc/openmandriva-release)"
+  else
+    OS=GNU/Linux
+  fi
+elif [ "x${GRUB_DISTRIBUTOR}" = "xopenmandriva" ] ; then
+  OS="$(sed 's, release .*$,,g' /etc/openmandriva-release)"
 else
-  OS="${GRUB_DISTRIBUTOR} GNU/Linux"
+  OS="${GRUB_DISTRIBUTOR}"
   CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr 'A-Z' 'a-z' | cut -d' ' -f1|LC_ALL=C sed 's,[^[:alnum:]_],_,g') ${CLASS}"
 fi
 
@@ -67,11 +73,20 @@
 
 case x"$GRUB_FS" in
     xbtrfs)
+	## (crazy) does not work with this FS
+	GRUB_SAVEDEFAULT=false
 	rootsubvol="`make_system_path_relative_to_its_root /`"
 	rootsubvol="${rootsubvol#/}"
 	if [ "x${rootsubvol}" != x ]; then
 	    GRUB_CMDLINE_LINUX="rootflags=subvol=${rootsubvol} ${GRUB_CMDLINE_LINUX}"
+		if [ "x${GRUB_DISABLE_RECOVERY}" != "xtrue" ]; then
+			GRUB_CMDLINE_LINUX_RECOVERY="rootflags=subvol=${rootsubvol} ${GRUB_CMDLINE_LINUX_RECOVERY}"
+		fi
 	fi;;
+    xf2fs)
+	## (crazy) does not work with this FS
+	GRUB_SAVEDEFAULT=false
+	;;
     xzfs)
 	rpool=`${grub_probe} --device ${GRUB_DEVICE} --target=fs_label 2>/dev/null || zdb -l ${GRUB_DEVICE} | awk -F \' '/ name/ { print $2 }'`
 	bootfs="`make_system_path_relative_to_its_root / | sed -e "s,@$,,"`"
@@ -97,12 +112,21 @@
   if [ -z "$boot_device_id" ]; then
       boot_device_id="$(grub_get_device_id "${GRUB_DEVICE}")"
   fi
+  # (crazy) do it that way so we have these translated
   if [ x$type != xsimple ] ; then
       case $type in
 	  recovery)
-	      title="$(gettext_printf "%s, with Linux %s (recovery mode)" "${os}" "${version}")"
+	      title_r="(recovery mode)"
+	      title_="$(gettext_printf "%s, with Linux %s" "${os}" "${version}")"
+	      title="${title_} ${title_r}"
 	      allowed_user_list="$(grub_get_auth_user_list $GRUB_RECOVERY_ACCESS_LEVEL)"
 	      ;;
+	  console)
+	      title_c="(console mode)"
+	      title_="$(gettext_printf "%s, with Linux %s" "${os}" "${version}")"
+	      title="${title_} ${title_c}"
+	      allowed_user_list="$(grub_get_auth_user_list $GRUB_ADVANCED_ACCESS_LEVEL)"
+	      ;;
 	  *)
 	      title="$(gettext_printf "%s, with Linux %s" "${os}" "${version}")"
 	      allowed_user_list="$(grub_get_auth_user_list $GRUB_ADVANCED_ACCESS_LEVEL)"
@@ -118,7 +142,7 @@
   else
       allowed_user_list="$(grub_get_auth_user_list $GRUB_SIMPLE_ACCESS_LEVEL)"
       echo "menuentry '$(echo "$os" | grub_quote)' ${CLASS} \$menuentry_id_option 'gnulinux-simple-$boot_device_id' ${allowed_user_list}{" | sed "s/^/$submenu_indentation/"
-  fi      
+  fi
   if [ x$type != xrecovery ] ; then
       save_default_entry | grub_add_tab
   fi
@@ -226,7 +250,7 @@
   version=`echo $basename | sed -e "s,^[^0-9]*-,,g"`
   alt_version=`echo $version | sed -e "s,\.old$,,g"`
   linux_root_device_thisversion="${LINUX_ROOT_DEVICE}"
-
+  #[[ -s /etc/machine-id ]] && machine_id=`cat /etc/machine-id`
   initrd_early=
   for i in ${GRUB_EARLY_INITRD_LINUX_STOCK} \
 	   ${GRUB_EARLY_INITRD_LINUX_CUSTOM}; do
@@ -305,16 +329,25 @@
     if [ -z "$boot_device_id" ]; then
 	boot_device_id="$(grub_get_device_id "${GRUB_DEVICE}")"
     fi
+
+    ## (crazy) this is bc the way translations are working
+    # match Live theme , requested from @bero
+    atext_c=">"
+    atext_="$(gettext_printf "Advanced options for %s" "${OS}" | grub_quote)"
+    atext="${atext_} ${atext_c}"
+
     # TRANSLATORS: %s is replaced with an OS name
-    echo "submenu '$(gettext_printf "Advanced options for %s" "${OS}" | grub_quote)' \$menuentry_id_option 'gnulinux-advanced-$boot_device_id' $(grub_echo_unrestricted_maybe){"
+    echo "submenu '$atext' \$menuentry_id_option 'gnulinux-advanced-$boot_device_id' $(grub_echo_unrestricted_maybe){"
     is_top_level=false
   fi
 
   linux_entry "${OS}" "${version}" advanced \
               "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT}"
+  linux_entry "${OS}" "${version}" console \
+              "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT} plymouth.enable=0 systemd.unit=multi-user.target"
   if [ "x${GRUB_DISABLE_RECOVERY}" != "xtrue" ]; then
     linux_entry "${OS}" "${version}" recovery \
-                "${GRUB_CMDLINE_LINUX_RECOVERY} ${GRUB_CMDLINE_LINUX}"
+                "${GRUB_CMDLINE_LINUX_RECOVERY} ${GRUB_CMDLINE_LINUX_RECOVERY}"
   fi
 done
 
